src/lib/utils.js

```diff
@@ -0,0 +1,20 @@
+// Utilities
+const queryString = require('query-string');
+
+/**
+ * Normalize querystring params. Remove empty array values
+ * @param {string} params
+ * @returns {string} Normalized querystring
+ */
+export function normalizeQueryStringParams(params) {
+    const qsParams = queryString.parse(params, { arrayFormat: 'index' });
+    const optionsCollection = Object.keys(qsParams).filter((key) => qsParams[key] instanceof Array);
+
+    optionsCollection.forEach((option) => {
+        const temp = qsParams[option].filter((item) => (item !== '' && item !== undefined));
+        qsParams[option] = temp;
+    });
+
+    const paramString = queryString.stringify(qsParams, { arrayFormat: 'index' });
```

Comment:
i'm wondering if we could get by with a `reduce` instead?  something like:

```javascript
const normalizedParams = Object.keys(qsParams, (memo, param) => {
  if (param instanceof Array) {
    memo[param] = param.filter(item => (item !== '' && item !== undefined));
  }
  return memo;
}, {});

Reply:
but this is a nitpick - just makes more sense to me that way

Reply:
i'd also like to get rid of that inner `filter`.  if it's okay that the items can be `0` or `null` we could do `params.filter((item === 0 || item === null) || item))` to exploit the truthiness of the value